feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go

```diff
@@ -0,0 +1,750 @@
+			Subscribe: &gnmipb.SubscriptionList{
+				Subscription: []*gnmipb.Subscription{
+					{
+						Path: &gnmipb.Path{
+							Elem: []*gnmipb.PathElem{{Name: "system"}, {Name: "state"}, {Name: "software-version"}},
+						},
+					},
+				},
+			},
+		},
+	}
+	t.Logf("Sending gNMI subscribe request: %s", sub)
+	err = stream.Send(sub)
+	if err != nil {
+		t.Logf("Failed to subscribe with error: %s", err)
+		return false
+	}
+	response, err := stream.Recv()
+	if err != nil {
+		if err != io.EOF {
+			t.Logf("Error received from the server: %s", err)
+			return false
+		}
+	}
+	t.Logf("gNMI response: %s", response)
+	conn.Close()
+	return true
+}
+
+// VerifyGribi function to validate the gRIBI service RPC after successful rotation.
+func VerifyGribi(t *testing.T, caCert *x509.CertPool, ctx context.Context, san, serverAddr, username, password string, cert tls.Certificate) bool {
+
+	credOpts := []grpc.DialOption{grpc.WithBlock(), grpc.WithTransportCredentials(credentials.NewTLS(
+		&tls.Config{
+			Certificates: []tls.Certificate{cert},
+			RootCAs:      caCert,
+			ServerName:   san,
+		}))}
+	creds := &rpcCredentials{&creds.UserPass{Username: username, Password: password}}
+	credOpts = append(credOpts, grpc.WithPerRPCCredentials(creds))
+	target := fmt.Sprintf("%s:%d", serverAddr, 9340)
+	conn, err := grpc.NewClient(target, credOpts...)
+	if err != nil {
+		t.Errorf("VerifyGnmi: gRPC NewClient failed to %q with error:%v", target, err)
+		return false
+	}
+	t.Logf("Connection state: %v.", conn.GetState().String())
+	defer conn.Close()
+
+	gRibiClient := gribipb.NewGRIBIClient(conn)
+	_, err = gRibiClient.Get(ctx, &gribipb.GetRequest{})
+	if err != nil {
+		t.Logf("Unable to connect GribiClient %v", err)
+		return false
+	}
+	conn.Close()
+	return true
+}
+
+// VerifyP4rt function to validate the P4rt service RPC after successful rotation.
+func VerifyP4rt(t *testing.T, caCert *x509.CertPool, ctx context.Context, san, serverAddr, username, password string, cert tls.Certificate) bool {
+
+	credOpts := []grpc.DialOption{grpc.WithBlock(), grpc.WithTransportCredentials(credentials.NewTLS(
+		&tls.Config{
+			Certificates: []tls.Certificate{cert},
+			RootCAs:      caCert,
+			ServerName:   san,
+		}))}
+	creds := &rpcCredentials{&creds.UserPass{Username: username, Password: password}}
+	credOpts = append(credOpts, grpc.WithPerRPCCredentials(creds))
+	target := fmt.Sprintf("%s:%d", serverAddr, 9559)
+	conn, err := grpc.NewClient(target, credOpts...)
+	if err != nil {
+		t.Errorf("VerifyP4rt : gRPC NewClient failed to %q", target)
+	}
+	t.Logf("Connection state: %v.", conn.GetState().String())
+	defer conn.Close()
+
+	p4RtClient := p4rtpb.NewP4RuntimeClient(conn)
+	_, err = p4RtClient.Capabilities(ctx, &p4rtpb.CapabilitiesRequest{})
+	if err != nil {
+		t.Logf("Unable to connect P4rtClient %v", err)
+		return false
+	}
+	conn.Close()
+	return true
+}
+
+// PreInitCheck function to do a validation of gNMI/gNOI/gRIBI/p4RT services before certz rotation.
```

Comment:
Can you explain why there is a difference here between the `PreInitCheck` and the `PostValidationCheck`? It seems like we need to have there is some opportunity here to have some commonality -- given that we'd expect that all services connect before and after.